finality-aleph/src/data_io/data_provider.rs

```diff
@@ -148,7 +148,15 @@ where
         // of `data_to_propose` unaffected.
 
         let client_info = self.client.info();
-        let finalized_block = (client_info.finalized_hash, client_info.finalized_number).into();
+        let finalized_block: BlockHashNum<B> =
```

Comment:
is ascription needed now?

Reply:
Yes :/

Reply:
okay, I see that it does not compile without it, what makes me really confused:
 - before it was not necessary (e.g. by usages like `prev.highest_finalized == finalized_block` the type could have been inferred)
 - it seems that because now the first usage (`finalized_block.num >= ...`) does not allow for a full inference the compiler starts shouting
 
strange :|